tracer/test/test-applications/integrations/Samples.InstrumentedTests/Vulnerabilities/EmailHtmlInjection/EmailHtmlInjectionAWSSDKSimpleEmailTests.cs

```diff
@@ -0,0 +1,164 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Net;
+using MimeKit;
+using Xunit;
+using Amazon.SimpleEmail;
+using Amazon.SimpleEmail.Model;
+using System.Web;
+using System;
+using Amazon.Runtime;
+using Moq;
+using System.Reflection;
+using MySql.Data.MySqlClient;
+using System.Threading.Tasks;
+using System.Threading;
+using Amazon;
+
+namespace Samples.InstrumentedTests.Iast.Vulnerabilities;
+
+public class EmailHtmlInjectionAWSSDKSimpleEmailTests : EmailInjectionBaseTests
+{
+    private MailboxAddress mailBoxSender = new MailboxAddress("sender", "address");
+    private MailboxAddress mailBoxRecipient = new MailboxAddress("recipient", "address");
+
+    // Test Task<SendEmailResponse> SendEmailAsync(SendEmailRequest request, System.Threading.CancellationToken cancellationToken = default(CancellationToken))
+
+    public class ClientMock: AmazonSimpleEmailServiceClient
+    {
+        public ClientMock()
+        { }
+    }
+
+    public class MyConfiguration: AmazonSimpleEmailServiceConfig
+    {
+        public override void Validate()
+        {
+            
+        }
+    }
+
+#if NETFRAMEWORK
+    [Fact]
+    public void GivenAnEmail_WhenSendHtmlMailMessageTaintedVaulesHtml_ThenIsVulnerable()
+    {
+        try
+        {
+            // The constructor of AmazonSimpleEmailServiceClient makes http calls that can make the test flaky and slow
+            // We can still test the aspect with a null object
+            ((AmazonSimpleEmailServiceClient)null).SendEmail(BuildMailMessage(true, taintedName, taintedLastName));
+        }
+        catch
+        {
+        }
+        AssertVulnerable();
+    }
+#endif
+
+    [Fact]
+    public void GivenAnEmail_WhenSendAsyncHtmlMailMessageTaintedVaulesHtml_ThenIsVulnerable()
+    {
+        try
+        {
+            // The constructor of AmazonSimpleEmailServiceClient makes http calls that can make the test flaky and slow
+            // We can still test the aspect with a null object
+            ((AmazonSimpleEmailServiceClient)null).SendEmailAsync(BuildMailMessage(true, taintedName, taintedLastName), default);
+        }
+        catch
+        {
+        }
+        AssertVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendAsyncHtmlMailMessageTaintedVaulesHtmlEscaped_ThenIsNotVulnerable()
+    {
+        try
+        {
+            ((AmazonSimpleEmailServiceClient)null).SendEmailAsync(BuildMailMessage(true, WebUtility.HtmlEncode(taintedName), WebUtility.HtmlEncode(taintedLastName)), default);
+        }
+        catch
+        {
+        }
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendAsyncHtmlMailMessageTaintedVaulesHtmlEscaped_ThenIsNotVulnerable2()
```

Comment:
same here, wondering if it could be factored as theories

Reply:
Done! Thanks!